DotNet 6 - VB - Error: The type or namespace name 'DeserializingBenchmarks' could not be found in the global namespace (are you missing an assembly reference?)

project file:
```
  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <OutputPath>..\..\..\bin\$(Configuration)\VB\Benchmarks\$(AssemblyName)\</OutputPath>
    <RootNamespace>DeserializingBenchmarks</RootNamespace>
    <TargetFramework>net6.0</TargetFramework>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="BenchmarkDotNet" Version="0.13.2" />
    <PackageReference Include="BenchmarkDotNet.Diagnostics.Windows" Version="0.13.2" />
  </ItemGroup>
```

Trimmed code used
```
Public Module Program

    Sub Main(args As String())

        Dim summary = BenchmarkRunner.Run(GetType(BenchmarkTests))

    End Sub


    <Orderer(SummaryOrderPolicy.FastestToSlowest)>
    <RankColumn(NumeralSystem.Arabic)>
    Public Class BenchmarkTests
   
        'benchmark code goes here...

    End Class
```

Reported summary header
```
// * Summary *

BenchmarkDotNet=v0.13.2, OS=Windows 11 (10.0.22000.1042/21H2)
AMD Ryzen 7 3700X, 1 CPU, 16 logical and 8 physical cores
.NET SDK=7.0.100-preview.7.22377.5
  [Host] : .NET 6.0.9 (6.0.922.41905), X64 RyuJIT AVX2
```

full error:
```
 Standard error:
 MSBuild version 17.4.0-preview-22368-02+c8492483a for .NET
C:\Program Files\dotnet\sdk\7.0.100-preview.7.22377.5\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.RuntimeIdentifierInference.targets(219,5): message NETSDK1057: You are using a preview version of .NET. See: https://aka.ms/dotnet-support-policy [C:\proj\projectName\bin\Release\VB\Benchmarks\DeserializingBenchmarks\net6.0\1774806d-717f-4cee-bb0c-1dcd40862f74\BenchmarkDotNet.Autogenerated.csproj]
C:\proj\projectName\bin\Release\VB\Benchmarks\DeserializingBenchmarks\net6.0\1774806d-717f-4cee-bb0c-1dcd40862f74\1774806d-717f-4cee-bb0c-1dcd40862f74.notcs(133,46): error CS0400: The type or namespace name 'DeserializingBenchmarks' could not be found in the global namespace (are you missing an assembly reference?) [C:\proj\projectName\bin\Release\VB\Benchmarks\DeserializingBenchmarks\net6.0\1774806d-717f-4cee-bb0c-1dcd40862f74\BenchmarkDotNet.Autogenerated.csproj]
C:\proj\projectName\bin\Release\VB\Benchmarks\DeserializingBenchmarks\net6.0\1774806d-717f-4cee-bb0c-1dcd40862f74\1774806d-717f-4cee-bb0c-1dcd40862f74.notcs(333,46): error CS0400: The type or namespace name 'DeserializingBenchmarks' could not be found in the global namespace (are you missing an assembly reference?) [C:\proj\projectName\bin\Release\VB\Benchmarks\DeserializingBenchmarks\net6.0\1774806d-717f-4cee-bb0c-1dcd40862f74\BenchmarkDotNet.Autogenerated.csproj]
C:\proj\projectName\bin\Release\VB\Benchmarks\DeserializingBenchmarks\net6.0\1774806d-717f-4cee-bb0c-1dcd40862f74\1774806d-717f-4cee-bb0c-1dcd40862f74.notcs(533,46): error CS0400: The type or namespace name 'DeserializingBenchmarks' could not be found in the global namespace (are you missing an assembly reference?) [C:\proj\projectName\bin\Release\VB\Benchmarks\DeserializingBenchmarks\net6.0\1774806d-717f-4cee-bb0c-1dcd40862f74\BenchmarkDotNet.Autogenerated.csproj]
C:\proj\projectName\bin\Release\VB\Benchmarks\DeserializingBenchmarks\net6.0\1774806d-717f-4cee-bb0c-1dcd40862f74\1774806d-717f-4cee-bb0c-1dcd40862f74.notcs(733,46): error CS0400: The type or namespace name 'DeserializingBenchmarks' could not be found in the global namespace (are you missing an assembly reference?) [C:\proj\projectName\bin\Release\VB\Benchmarks\DeserializingBenchmarks\net6.0\1774806d-717f-4cee-bb0c-1dcd40862f74\BenchmarkDotNet.Autogenerated.csproj]
C:\proj\projectName\bin\Release\VB\Benchmarks\DeserializingBenchmarks\net6.0\1774806d-717f-4cee-bb0c-1dcd40862f74\1774806d-717f-4cee-bb0c-1dcd40862f74.notcs(933,46): error CS0400: The type or namespace name 'DeserializingBenchmarks' could not be found in the global namespace (are you missing an assembly reference?) [C:\proj\projectName\bin\Release\VB\Benchmarks\DeserializingBenchmarks\net6.0\1774806d-717f-4cee-bb0c-1dcd40862f74\BenchmarkDotNet.Autogenerated.csproj]
C:\proj\projectName\bin\Release\VB\Benchmarks\DeserializingBenchmarks\net6.0\1774806d-717f-4cee-bb0c-1dcd40862f74\1774806d-717f-4cee-bb0c-1dcd40862f74.notcs(1133,46): error CS0400: The type or namespace name 'DeserializingBenchmarks' could not be found in the global namespace (are you missing an assembly reference?) [C:\proj\projectName\bin\Release\VB\Benchmarks\DeserializingBenchmarks\net6.0\1774806d-717f-4cee-bb0c-1dcd40862f74\BenchmarkDotNet.Autogenerated.csproj]
C:\proj\projectName\bin\Release\VB\Benchmarks\DeserializingBenchmarks\net6.0\1774806d-717f-4cee-bb0c-1dcd40862f74\1774806d-717f-4cee-bb0c-1dcd40862f74.notcs(1333,46): error CS0400: The type or namespace name 'DeserializingBenchmarks' could not be found in the global namespace (are you missing an assembly reference?) [C:\proj\projectName\bin\Release\VB\Benchmarks\DeserializingBenchmarks\net6.0\1774806d-717f-4cee-bb0c-1dcd40862f74\BenchmarkDotNet.Autogenerated.csproj]
Build FAILED.
C:\proj\projectName\bin\Release\VB\Benchmarks\DeserializingBenchmarks\net6.0\1774806d-717f-4cee-bb0c-1dcd40862f74\1774806d-717f-4cee-bb0c-1dcd40862f74.notcs(133,46): error CS0400: The type or namespace name 'DeserializingBenchmarks' could not be found in the global namespace (are you missing an assembly reference?) [C:\proj\projectName\bin\Release\VB\Benchmarks\DeserializingBenchmarks\net6.0\1774806d-717f-4cee-bb0c-1dcd40862f74\BenchmarkDotNet.Autogenerated.csproj]
C:\proj\projectName\bin\Release\VB\Benchmarks\DeserializingBenchmarks\net6.0\1774806d-717f-4cee-bb0c-1dcd40862f74\1774806d-717f-4cee-bb0c-1dcd40862f74.notcs(333,46): error CS0400: The type or namespace name 'DeserializingBenchmarks' could not be found in the global namespace (are you missing an assembly reference?) [C:\proj\projectName\bin\Release\VB\Benchmarks\DeserializingBenchmarks\net6.0\1774806d-717f-4cee-bb0c-1dcd40862f74\BenchmarkDotNet.Autogenerated.csproj]
C:\proj\projectName\bin\Release\VB\Benchmarks\DeserializingBenchmarks\net6.0\1774806d-717f-4cee-bb0c-1dcd40862f74\1774806d-717f-4cee-bb0c-1dcd40862f74.notcs(533,46): error CS0400: The type or namespace name 'DeserializingBenchmarks' could not be found in the global namespace (are you missing an assembly reference?) [C:\proj\projectName\bin\Release\VB\Benchmarks\DeserializingBenchmarks\net6.0\1774806d-717f-4cee-bb0c-1dcd40862f74\BenchmarkDotNet.Autogenerated.csproj]
C:\proj\projectName\bin\Release\VB\Benchmarks\DeserializingBenchmarks\net6.0\1774806d-717f-4cee-bb0c-1dcd40862f74\1774806d-717f-4cee-bb0c-1dcd40862f74.notcs(733,46): error CS0400: The type or namespace name 'DeserializingBenchmarks' could not be found in the global namespace (are you missing an assembly reference?) [C:\proj\projectName\bin\Release\VB\Benchmarks\DeserializingBenchmarks\net6.0\1774806d-717f-4cee-bb0c-1dcd40862f74\BenchmarkDotNet.Autogenerated.csproj]
C:\proj\projectName\bin\Release\VB\Benchmarks\DeserializingBenchmarks\net6.0\1774806d-717f-4cee-bb0c-1dcd40862f74\1774806d-717f-4cee-bb0c-1dcd40862f74.notcs(933,46): error CS0400: The type or namespace name 'DeserializingBenchmarks' could not be found in the global namespace (are you missing an assembly reference?) [C:\proj\projectName\bin\Release\VB\Benchmarks\DeserializingBenchmarks\net6.0\1774806d-717f-4cee-bb0c-1dcd40862f74\BenchmarkDotNet.Autogenerated.csproj]
C:\proj\projectName\bin\Release\VB\Benchmarks\DeserializingBenchmarks\net6.0\1774806d-717f-4cee-bb0c-1dcd40862f74\1774806d-717f-4cee-bb0c-1dcd40862f74.notcs(1133,46): error CS0400: The type or namespace name 'DeserializingBenchmarks' could not be found in the global namespace (are you missing an assembly reference?) [C:\proj\projectName\bin\Release\VB\Benchmarks\DeserializingBenchmarks\net6.0\1774806d-717f-4cee-bb0c-1dcd40862f74\BenchmarkDotNet.Autogenerated.csproj]
C:\proj\projectName\bin\Release\VB\Benchmarks\DeserializingBenchmarks\net6.0\1774806d-717f-4cee-bb0c-1dcd40862f74\1774806d-717f-4cee-bb0c-1dcd40862f74.notcs(1333,46): error CS0400: The type or namespace name 'DeserializingBenchmarks' could not be found in the global namespace (are you missing an assembly reference?) [C:\proj\projectName\bin\Release\VB\Benchmarks\DeserializingBenchmarks\net6.0\1774806d-717f-4cee-bb0c-1dcd40862f74\BenchmarkDotNet.Autogenerated.csproj]
    0 Warning(s)
    7 Error(s)
```

--- a/src/BenchmarkDotNet/Toolchains/CsProj/CsProjGenerator.cs
+++ b/src/BenchmarkDotNet/Toolchains/CsProj/CsProjGenerator.cs
@@ -154,19 +154,24 @@ namespace BenchmarkDotNet.Toolchains.CsProj
             // important assumption! project's file name === output dll name
             string projectName = benchmarkTarget.GetTypeInfo().Assembly.GetName().Name;
 
-            // I was afraid of using .GetFiles with some smart search pattern due to the fact that the method was designed for Windows
-            // and now .NET is cross platform so who knows if the pattern would be supported for other OSes
             var possibleNames = new HashSet<string> { $"{projectName}.csproj", $"{projectName}.fsproj", $"{projectName}.vbproj" };
-            var projectFile = rootDirectory
-                .EnumerateFiles("*.*", SearchOption.AllDirectories)
-                .FirstOrDefault(file => possibleNames.Contains(file.Name));
+            var projectFiles = rootDirectory
+                .EnumerateFiles("*proj", SearchOption.AllDirectories)
+                .Where(file => possibleNames.Contains(file.Name))
+                .ToArray();
 
-            if (projectFile == default(FileInfo))
+            if (projectFiles.Length == 0)
             {
                 throw new NotSupportedException(
                     $"Unable to find {projectName} in {rootDirectory.FullName} and its subfolders. Most probably the name of output exe is different than the name of the .(c/f)sproj");
             }
-            return projectFile;
+            else if (projectFiles.Length > 1)
+            {
+                throw new NotSupportedException(
+                    $"Found more than one matching project file for {projectName} in {rootDirectory.FullName} and its subfolders: {string.Join(",", projectFiles.Select(pf => $"'{pf.FullName}'"))}. Benchmark project names needs to be unique.");
+            }
+
+            return projectFiles[0];
         }
 
         public override bool Equals(object obj) => obj is CsProjGenerator other && Equals(other);

